text/0000-subslice-offset.md

```diff
@@ -0,0 +1,344 @@
+- Feature Name: `subslice_offset`
+- Start Date: 2019-10-27
+- RFC PR: None yet
+- Rust Issue: None yet
+
+# Summary
+[summary]: #summary
+
+Add functions to the slice primitive to get the offset/index of an element or
+subslice based on their memory addresses:
+
+```rust
+impl<T> [T] {
+    pub fn index_of(&self, element: &T) -> Option<usize>;
+    pub fn range_of(&self, subslice: &[T]) -> Option<Range<usize>>;
```

Comment:
Returning a range instead of just the start offset seems not strictly necessary since its length should always be `subslice.len()`, but on further thought it’s nice to have and makes the meaning of the return value obvious. So :+1: